File: src/int_max.cpp
#include <climits>
#include <iostream>

using namespace std; 

int main ( int argc , char * argv[])

{
 
  cout << "INT_MAX      " << INT_MAX   << endl ;
  cout << "INT_MAX +1 = " << INT_MAX + 1  << endl ;
  cout << "INT_MAX -1 = " << INT_MAX - 1 << endl ;
 
  cout << "INT_MAX / INT_MAX        " << INT_MAX /INT_MAX << endl ;
  cout << "(INT_MAX +1) / INT_MAX   " << (INT_MAX +1) /INT_MAX << endl;
  cout << "(INT_MAX -1) / INT_MAX   " << (INT_MAX -1) /INT_MAX  <<endl;
  cout << "INT_MAX / (INT_MAX +1)   " << INT_MAX  /(INT_MAX+1)  <<endl;
  cout << "INT_MAX / (INT_MAX -1)   " << INT_MAX  /(INT_MAX -1)  <<endl;

}

File: src/IntMax.java
public class IntMax {
    
    public static void main( String[] args ) {
        
        System.out.println("Integer.MAX_VALUE = " + Integer.MAX_VALUE);
        System.out.println("Integer.MAX_VALUE + 1 = " + (Integer.MAX_VALUE+1));
        System.out.println("Integer.MAX_VALUE + 1 = " + (Integer.MAX_VALUE-1));
        
        System.out.println("");
        
        System.out.println("Integer.MAX_VALUE / Integer.MAX_VALUE = " + (Integer.MAX_VALUE/Integer.MAX_VALUE));
        System.out.println("(Integer.MAX_VALUE + 1) / Integer.MAX_VALUE = " + ((Integer.MAX_VALUE+1)/Integer.MAX_VALUE));
        System.out.println("(Integer.MAX_VALUE - 1) / Integer.MAX_VALUE = " + ((Integer.MAX_VALUE-1)/Integer.MAX_VALUE));
        System.out.println("Integer.MAX_VALUE / (Integer.MAX_VALUE + 1) = " + (Integer.MAX_VALUE/(Integer.MAX_VALUE+1)));
        System.out.println("Integer.MAX_VALUE / (Integer.MAX_VALUE - 1) = " + (Integer.MAX_VALUE/(Integer.MAX_VALUE-1)));
        
    }
    
}